### PR TITLE
Replace uri with prefixes for turtle format

### PR DESCRIPTION
**Pull request by rumilbaybikov**

### CODE REVIEW

File: lib/bel/translator/plugins/rdf/translator.rb [hunk 1]
```diff
@@ -94,6 +94,16 @@ def write_rdf_to_io(format, objects, io, options = {})
         :stream => true
       )
 
+      if (@format == :turtle)
```
Comment (abargnesi) [inline lib/bel/translator/plugins/rdf/translator.rb]: We shouldn't need to restrict these prefixes to only the `:turtle` format. Let's remove this condition so other RDF formats that support prefixes will use the YAML definitions.

File: lib/bel/translator/plugins/rdf/translator.rb [hunk 2]
```diff
@@ -94,6 +94,16 @@ def write_rdf_to_io(format, objects, io, options = {})
...
+        require 'yaml'
```
Comment (abargnesi) [inline lib/bel/translator/plugins/rdf/translator.rb]: Move the require to the top of the file. If we remove the condition above then we'll always need to load the yaml module.

File: lib/bel/translator/plugins/rdf/translator.rb [hunk 3]
```diff
@@ -94,6 +94,16 @@ def write_rdf_to_io(format, objects, io, options = {})
...
+
+        prefixes = YAML::load_file('config/namespace_prefixes.yml')
```
Comment (abargnesi) [inline lib/bel/translator/plugins/rdf/translator.rb]: The `config/namespace_prefixes.yml` is a relative path. This code will expect to find it relative to the working directory when you run the `bel` command.

For example if I install the gem with your PR changes and translate bel to turtle in a directory where `config/namespace_prefixes.yml` does not exist, I will receive this error:

```bash
[tony@starship openbel]$ pwd
/home/tony/projects/openbel
[tony@starship openbel]$ bel translate -i bel.rb/small_corpus.bel bel turtle
[DEPRECATION] the SKOS vocabulary will be moved to the rdf-vocab gem
for the RDF.rb 2.0 release. Use as RDF::Vocab::SKOS, or include RDF::Vocab in the RDF module.
Called from /home/tony/.gem/ruby/2.3.0/gems/bel-0.6.0/lib/bel/translator/plugins/rdf/bel_schema.rb:197

/usr/lib/ruby/2.3.0/psych.rb:471:in `initialize': No such file or directory @ rb_sysopen - config/namespace_prefixes.yml (Errno::ENOENT)
	from /usr/lib/ruby/2.3.0/psych.rb:471:in `open'
	from /usr/lib/ruby/2.3.0/psych.rb:471:in `load_file'
	from /home/tony/.gem/ruby/2.3.0/gems/bel-0.6.0/lib/bel/translator/plugins/rdf/translator.rb:100:in `write_rdf_to_io'
	from /home/tony/.gem/ruby/2.3.0/gems/bel-0.6.0/lib/bel/translator/plugins/rdf/translator.rb:31:in `write'
	from /home/tony/.gem/ruby/2.3.0/gems/bel-0.6.0/lib/bel/translate.rb:53:in `translate'
	from /home/tony/.gem/ruby/2.3.0/gems/bel-0.6.0/bin/bel:304:in `run'
	from /home/tony/.gem/ruby/2.3.0/gems/bel-0.6.0/bin/bel:516:in `<top (required)>'
	from /home/tony/.gem/ruby/2.3.0/bin/bel:23:in `load'
	from /home/tony/.gem/ruby/2.3.0/bin/bel:23:in `<main>'
```

Instead maybe we can put `config/namespace_prefixes.yml` under the `$projectRoot/lib/bel/translator/plugins/rdf` path and load it relative to the translator.rb file.

File: lib/bel/translator/plugins/rdf/translator.rb [hunk 4]
```diff
@@ -94,6 +94,16 @@ def write_rdf_to_io(format, objects, io, options = {})
...
+
+        prefixes.each do |prefix, uri|
+          rdf_writer.prefix prefix.to_sym, RDF::URI(uri)
```
Comment (abargnesi) [inline lib/bel/translator/plugins/rdf/translator.rb]: This looks great!

**Comment by abargnesi** [comment]
@rumilbaybikov The relative path resolution seems the most problematic. Is it intended that this configuration file be provided by the user or included within the gem?

File: lib/bel/translator/plugins/rdf/translator.rb (hunk 1 above)
Comment (rumilbaybikov) [inline lib/bel/translator/plugins/rdf/translator.rb]: ok

File: lib/bel/translator/plugins/rdf/translator.rb (hunk 2 above)
Comment (rumilbaybikov) [inline lib/bel/translator/plugins/rdf/translator.rb]: ok

File: lib/bel/translator/plugins/rdf/translator.rb (hunk 3 above)
Comment (rumilbaybikov) [inline lib/bel/translator/plugins/rdf/translator.rb]: Resolved using the Pathname class: Pathname.new('config/namespace_prefixes.yml')

**Comment by rumilbaybikov** [comment]
@abargnesi Added a new option -p containing the path to custom user configuration file with prefixes. If option -p is not provided then default prefixes defined in config/namespaces_prefixes.yml will be used.

Usage:
bel translate -i bel.rb/small_corpus.bel -p bel.rb/prefixes.yml bel turtle

**Comment by abargnesi** [comment]
Thanks @rumilbaybikov!

I merged manually due to conflicts applying the second commit. It is generally recommended to put work on a separate *topic branch* in git. This branch should then be offered in the PR.

Consult the [contributing guide](https://github.com/OpenBEL/bel.rb/blob/master/CONTRIBUTING.md) for details on branches in bel.rb. Also check out this [terminalcast](https://asciinema.org/a/40097) for an example of making a change on a *topic branch*.

**Comment by abargnesi** [comment]
This has been merged back to the *next* branch in a6cf83b.

**Comment by abargnesi** [comment]
@rumilbaybikov, @juliakozlovsky

Please read and fill out a CLA (individual or corporate) and submit to the [OpenBEL/cla](https://github.com/OpenBEL/cla) project. This is required once.

See [here](https://github.com/OpenBEL/cla/blob/master/CONTRIBUTING.adoc) for steps.

**Comment by juliakozlovsky** [comment]
Hi Tony - I spoke to our (mine, Alexander's and Rumil's) manager, and he said he needs to run this through our Legal people, and it needs to be corporate-level thing from our end, not individual.  We don't currently anticipate any specific issues with doing this per se, but we need some time to sort out our company policy.

**Pull request merged**